apps/daimo-mobile/src/action/actStatus.ts

```diff
@@ -0,0 +1,38 @@
+import { useCallback, useState } from "react";
+
+type ActStatus = "idle" | "loading" | "success" | "error";
+
+export type SetActStatus = (
+  status: ActStatus | Error,
+  message?: string
+) => void;
+
+export interface ActHandle {
+  /** Action status */
+  status: ActStatus;
+  /** Empty when idle. Describes progress, success, or failure. */
+  message: string;
+  /** Should be called only when status is 'idle' */
+  exec: () => void;
+}
+
+/** Tracks progress of a user action. */
+export function useActStatus() {
```

Comment:
this is a nice reusable pattern

react hook which is meant to track a single user action, such as
- send
- add device 
- remove device
- create note
- redeem note

Those are all onchain ops. Pattern would fit future other ops (like waiting for approval from an onramp) as well.

- status `idle` > `loading` > `success`/`failure`
- at each point, a short updating message to show granular progress

i think we can use this hook for telemetry, to track anonymized / rolled-up performance and reliability data